lib/ansible/modules/remote_management/ucs/ucs_ip_pool.py

```diff
@@ -0,0 +1,294 @@
+    default: '64'
+  ipv6_default_gw:
+    description:
+    - The default gateway associated with the IPv6 addresses in the block.
+    default: '::'
+  ipv6_primary_dns:
+    description:
+    - The primary DNS server that this block of IPv6 addresses should access.
+    default: '::'
+  ipv6_secondary_dns:
+    description:
+    - The secondary DNS server that this block of IPv6 addresses should access.
+    default: '::'
+  org_dn:
+    description:
+    - Org dn (distinguished name)
+    default: org-root
+requirements:
+- ucsmsdk
+author:
+- David Soper (@dsoper2)
+- CiscoUcs (@CiscoUcs)
+version_added: '2.5'
+'''
+
+EXAMPLES = r'''
+- name: Configure IPv4 and IPv6 address pools
+  ucs_ip_pool:
+    hostname: 172.16.143.150
+    username: admin
+    password: password
+    name: ip-A
+    order: sequential
+    first_addr: 192.168.0.10
+    last_addr: 192.168.0.19
+    subnet_mask: 255.255.255.0
+    default_gw: 192.168.0.1
+    primary_dns: 172.16.143.136
+  ucs_ip_pool:
+    hostname: 172.16.143.150
+    username: admin
+    password: password
+    name: ipv6-B
+    ipv6_first_addr: fe80::1cae:7992:d7a1:ed07
+    ipv6_last_addr: fe80::1cae:7992:d7a1:edfe
+    ipv6_default_gw: fe80::1cae:7992:d7a1:ecff
+
+- name: Remove IPv4 and IPv6 address pools
+  ucs_ip_pool:
+    hostname: 172.16.143.150
+    username: admin
+    password: password
+    name: ip-A
+    state: absent
+  ucs_ip_pool:
+    hostname: 172.16.143.150
+    username: admin
+    password: password
+    name: ipv6-B
+    state: absent
+'''
+
+RETURN = r'''
+#
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.remote_management.ucs import UCSModule, ucs_argument_spec
+
+
+def main():
+    argument_spec = ucs_argument_spec
+    argument_spec.update(
+        org_dn=dict(type='str', default='org-root'),
+        name=dict(type='str', required=True),
+        descr=dict(type='str', default=''),
+        order=dict(type='str', default='default', choices=['default', 'sequential']),
+        first_addr=dict(type='str'),
+        last_addr=dict(type='str'),
+        subnet_mask=dict(type='str', default='255.255.255.0'),
+        default_gw=dict(type='str', default='0.0.0.0'),
+        primary_dns=dict(type='str', default='0.0.0.0'),
+        secondary_dns=dict(type='str', default='0.0.0.0'),
+        ipv6_first_addr=dict(type='str'),
+        ipv6_last_addr=dict(type='str'),
+        ipv6_prefix=dict(type='str', default='64'),
+        ipv6_default_gw=dict(type='str', default='::'),
+        ipv6_primary_dns=dict(type='str', default='::'),
+        ipv6_secondary_dns=dict(type='str', default='::'),
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+    )
+
+    module = AnsibleModule(
+        argument_spec,
+        supports_check_mode=True,
+    )
+    ucs = UCSModule(module)
+
+    err = False
+
+    from ucsmsdk.mometa.ippool.IppoolPool import IppoolPool
```

Comment:
We expect all imports to happen at the start of the module.
If it is possible this will fail because e.g. ucsmsdk is missing, then you add it as part of a try/except block and set HAS_UCSMSDK=True/False, and provide a proper fail_json error if HAS_UCSMSDK is False.